### ISSUE TITLE
Issues with React

### ISSUE THREAD

**Issue by aklinker1** [aklinker1/vite-plugin-web-extension#139]
### Summary

React has problems. If a component isn't a default, anonymous export, the extension won't run in dev mode correctly. It will either log error messages, or HMR/fast reload just won't work. See https://github.com/aklinker1/vite-plugin-web-extension/issues/136#issuecomment-1685371260 for current behavior.

This means large react application will be difficult, if not impossible, to write using this plugin.

###### This works

```ts
export default function () {
  return <div />
}
```

###### This doesn't

```ts
export const App = () => {
  return <div />
}
```

Also see:

- #136 
- #117 
- #84 
- #114 

### Reproduction

See #136 for an example reproduction.

**Comment by FarazFKhan** [comment]
Could someone please explain the difference between the "vite-plugin-web-extension" and the "wxt" web extension frameworks?

**Comment by aklinker1** [comment]
@FarazFKhan There's a TLDR at the bottom.

The main difference is that `vite-plugin-web-extension` is a vite plugin that operates with Vite's limited plugin lifecycle, whereas WXT is a framework build on top of Vite with additional functionality outside the build process.

<img width="749" alt="Screenshot 2023-08-21 at 7 15 57 PM" src="https://github.com/aklinker1/vite-plugin-web-extension/assets/10101283/ca426b98-fd7e-44fc-99b0-4241a928cbd2">

Web extensions, with their multiple build steps (one per entrypoint), need to schedule multiple Vite builds to output a fully functioning extension. Making major modifications like this to the build process from inside a plugin is difficult to do and easily breaks with non-breaking Vite changes.

`vite-plugin-web-extension` orchestrates all the required builds for a web extension by blocking the "main" build until all these "child" builds are completed.

```
vite build (no vite-plugin-web-extension)
└─ vite

vite build (with vite-plugin-web-extension)
└─ vite (main)
   ├─ vite (popup, options, other html pages...)
   ├─ vite (sandbox pages...)
   ├─ vite (background)
   ├─ vite (content script 1)
   ├─ vite (content script 2)
   └─ vite (content script ...)
```

When you consider dev mode and the dev server, things get more complicated. Different lifecycle events fire during dev mode, meaning we need to orchestrate the builds in a completely separate code path, but with the same lifecycle restrictions.

Ideally, when building a web extension, we need to be able to work outside the Vite's "existing build process", and plugins don't allow us to do that.

Ok, so what about WXT? WXT is a framework that uses Vite as a build tool internally. 

The build process WXT uses is similar to Vite, but simpler:

```
wxt build
├─ vite (popup, options, other html pages...)
├─ vite (sandbox pages...)
├─ vite (background)
├─ vite (content script 1)
├─ vite (content script 2)
└─ vite (content script ...)
```

Since WXT controls vite, it can eliminate the "main" vite build (normally started when running `vite build`), which restricts when we can run code. It lets us leverage new parts of the build cycle, like before and after all builds. Plugin's don't have hooks for this.

### TLDR

I made WXT because when you need to orchestrate multiple builds, a plugin is not the correct solution. You need to be in charge of the entire build process from start to finish. This is also what the Vite team recommends.

> I'll add a link here when I find the source for this statement, it's on an issue somewhere that I read a while back.

In addition to simplifying the build step, a framework must come with additional features. WXT includes auto-imports, zipping and publishing utils, improved typing to the `browser` object, simpler inline configuration, and more. That is what makes WXT a framework, while this plugin is not.
